### PR TITLE
FIX/LIVE-15117: exchange.swap does not have up-to-date accounts list when attempting to swap with newly created account.

### PR DESCRIPTION
**Pull request by kallen-ledger**
### ✅ Checklist

- [x] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description
- When creating a new account on the swap live app interface.
- The user will attempt to swap "to" this new account.
- The user will then be unable to.

This was because the `exchangeHandlers` accounts param did not get updated when accounts were changed. Instead this PR proposes passing a getter function that will instead grab the users currents accounts when the function is run. 

### ❓ Context

https://ledgerhq.atlassian.net/browse/LIVE-15117

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by CremaFR — DISMISSED** [review]
LGTM 🔥

**Review by liviuciulinaru — DISMISSED** [review]

File: apps/ledger-live-desktop/src/renderer/init.tsx
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/init.tsx:1]: Is this change necessary ?
It doesn't seem correct to init the rest and not this in the function

File: apps/ledger-live-desktop/src/renderer/components/WebPTXPlayer/CustomHandlers.ts [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
 import { context } from "~/renderer/drawers/Provider";
 import WebviewErrorDrawer from "~/renderer/screens/exchange/Swap2/Form/WebviewErrorDrawer";
 import { platformAppDrawerStateSelector } from "~/renderer/reducers/UI";
+import { flattenAccountsSelector } from "~/renderer/reducers/accounts";
+import { store } from "~/renderer/init";
 
-export function usePTXCustomHandlers(manifest: WebviewProps["manifest"], accounts: AccountLike[]) {
+function getAccounts() {
+  return flattenAccountsSelector(store.getState());
+}
```
Comment (Justkant) [inline apps/ledger-live-desktop/src/renderer/components/WebPTXPlayer/CustomHandlers.ts:22]: It's quite strange to do that directly instead of using a the selector with `useSelector`
It's seems we're not addressing the root cause of the issue, a change to the accounts should be reflected to the handlers

**Review by Justkant — COMMENTED** [review]
I feel like this solution is bypassing the issue instead of solving it

File: apps/ledger-live-desktop/src/renderer/init.tsx [hunk 2]
```diff
@@ -55,6 +55,10 @@ import { registerTransportModules } from "~/renderer/live-common-setup";
 const rootNode = document.getElementById("react-root");
 const TAB_KEY = 9;
 
+export const store = createStore({
```
[the remark this answers is not in the document]
Comment (kallen-ledger) [reply apps/ledger-live-desktop/src/renderer/init.tsx:58]: As far as I can tell this does not create any side effects and just moves the creation of the store outside of the init function to allow access to the store directly.

**Comment by kallen-ledger** [comment]
@Justkant @KVNLS agree that it is not solving the core issue, will take a re-look at this and try and figure out why the accounts are not getting updated on call. Thank you for the feedback 🙏

**Review by CremaFR — APPROVED** [review]

**Review by Justkant — APPROVED** [review]
